### ISSUE TITLE
"pg_ctl: could not start server" after huge_pages workaround

### ISSUE THREAD

**Issue by ntx-ln** [bitnami/containers#7890]
### Name and Version

bitnami/postgresql:14

### What steps will reproduce the bug?

1. Running a microk8s with mayastor enabled and an 128GB block device as msp
2. `k create ns lntest`
3. `k -n lntest apply -f ./ln-postgres.yaml`
4. `k -n lntest logs pod/ln-postgres`

### What is the expected behavior?

The postgres-DB starting up.

### What do you see instead?
```
postgresql 13:04:13.83 
postgresql 13:04:13.83 Welcome to the Bitnami postgresql container
postgresql 13:04:13.83 Subscribe to project updates by watching https://github.com/bitnami/containers
postgresql 13:04:13.83 Submit issues and feature requests at https://github.com/bitnami/containers/issues
postgresql 13:04:13.84 
postgresql 13:04:13.84 DEBUG ==> Configuring libnss_wrapper...
postgresql 13:04:13.88 INFO  ==> ** Starting PostgreSQL setup **
postgresql 13:04:13.91 INFO  ==> Validating settings in POSTGRESQL_* env vars..
postgresql 13:04:13.91 WARN  ==> You set the environment variable ALLOW_EMPTY_PASSWORD=yes. For safety reasons, do not use this flag in a production environment.
postgresql 13:04:13.92 INFO  ==> Loading custom pre-init scripts...
postgresql 13:04:13.93 INFO  ==> Initializing PostgreSQL database...
postgresql 13:04:13.93 DEBUG ==> Copying files from /bitnami/postgresql/conf to /opt/bitnami/postgresql/conf
postgresql 13:04:13.94 DEBUG ==> Ensuring expected directories/files exist...
postgresql 13:04:13.97 INFO  ==> pg_hba.conf file not detected. Generating it...
postgresql 13:04:13.97 INFO  ==> Generating local authentication configuration
The files belonging to this database system will be owned by user "postgres".
This user must also own the server process.

The database cluster will be initialized with locale "en_US.UTF-8".
The default text search configuration will be set to "english".

Data page checksums are disabled.

fixing permissions on existing directory /bitnami/postgresql/data ... ok
creating subdirectories ... ok
selecting dynamic shared memory implementation ... posix
selecting default max_connections ... 20
selecting default shared_buffers ... 400kB
selecting default time zone ... Etc/UTC
creating configuration files ... ok
running bootstrap script ... ok
performing post-bootstrap initialization ... ok
initdb: warning: enabling "trust" authentication for local connections
You can change this by editing pg_hba.conf or using the option -A, or
--auth-local and --auth-host, the next time you run initdb.
syncing data to disk ... ok


Success. You can now start the database server using:

    /opt/bitnami/postgresql/bin/pg_ctl -D /bitnami/postgresql/data -l logfile start

postgresql 13:04:23.19 INFO  ==> Starting PostgreSQL in background...
waiting for server to start.... stopped waiting
pg_ctl: could not start server
Examine the log output.
```
### Additional information

ln-postgres.yaml
```yaml
apiVersion: v1
kind: Pod
metadata:
  name: ln-postgres
  labels:
    name: ln-postgres
spec:
  containers:
  - name: ln-postgres
    image: docker.io/bitnami/postgresql:14
    volumeMounts:
    - name: postgresql-config
      mountPath: /opt/bitnami/postgresql/share/postgresql.conf.sample
      subPath: postgresql.conf.sample
    env:
    - name: ALLOW_EMPTY_PASSWORD
      value: "yes"
    - name: BITNAMI_DEBUG
      value: "true"
  volumes:
  - name: postgresql-config
    configMap:
      name: postgresql-cm  
---
apiVersion: v1
kind: ConfigMap
metadata:
  name: postgresql-cm
data:
  postgresql.conf.sample: | 
    huge_pages = off                       	# on, off, or try
```

I am mounting this cm because mayastor requires huge_pages, which postgres does not seem to function with on my system. I disable them this way as suggested in: https://github.com/bitnami/charts/issues/7901#issuecomment-970002063
EDIT: I relized pretty much everything in my cm was a comment, so I just kept the one line that wasnt.

I can rule out liveness- and readiness-probes, as I pushed the initial delay to 900s for both of them and the pod still crashed pretty much instant.

**Comment by gongomgra** [comment]
Hi @ntx-ln,

Sorry for the delay. I'm afraid I don't have a test environment with the `huge_pages` setting enabled, but I think I understand your problem. 

Apart from setting the `huge_pages = 'off'` (please use the single quote marks here just in case) for the init process with the `postgresql.conf.sample` file, can you also set it in the final `postgresql.conf` file for the setting to be used at runtime? Using our chart, you can configure it in the `primary.extendedConfiguration` parameter in the `values.yaml` file as below. You can also configure the custom config map for the `postgresql.conf.sample` file in the `primary.existingConfigmap` field.

```
extendedConfiguration: "huge_pages = 'off'"
```

In my case, the pod started just fine, and I checked the PostgreSQL process is running with `huges_pages = off` setting

```
$ psql -U postgres -d postgres
Password for user postgres:
psql (14.5)
Type "help" for help.

postgres=# show huge_pages;
 huge_pages
------------
 off
(1 row)

postgres=# exit
```

**Comment by ntx-ln** [comment]
Hey @gongomgra,
thank you for your answer! I was on a leave last week, that's why I answer so late. During my absence my boss saw that I mounted the config on conf.sample and he changed that to the normal conf file. I did not try that before, because I read in a different Issue, that the .conf.sample is for some reason used instead of the .conf. That did help however and our postgres container is running now.
Thanks again!
